cmd/lakectl/cmd/root_test.go

```diff
@@ -133,27 +120,24 @@ func TestInitConfig_LoadingScenarios(t *testing.T) {
 
 				cfgFile = tmpFileFlag
 				t.Setenv("LAKECTL_CONFIG_FILE", tmpFileEnv)
-
-				return func() {
-					cfgFile = ""
-					os.Unsetenv("LAKECTL_CONFIG_FILE")
-				}
 			},
 			expectedURL: "http://from-flag",
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			cfgFile = ""
 			viper.Reset()
 
-			cleanup := tt.setup(t)
-			defer cleanup()
+			originalCfgFile := cfgFile
+			defer func() { cfgFile = originalCfgFile }()
+
+			cfgFile = ""
+
+			tt.setup(t)
 
 			initConfig()
 			require.NoError(t, viper.Unmarshal(&cfg), "Failed to unmarshal config")
```

Comment:
suggest not to update the global `cfg` and have a another struct to initialize in the test scope for testing.